### ISSUE TITLE
singularity recipe :: proposal

### ISSUE THREAD

**Issue by EricDeveaud** [wheaton5/souporcell#47]
Hello, 

one of our cluster still run in centos, so we are stuck to singularity containers running  a glibc <= 2.25 to be able to be executed.
so this means basically 

- centos6 based
- ubuntu-16.04

so I add to rewrite the `singularity.def` file in order to accomodate this problem, 
I based  this recipe on `nvcr.io/nvidia/cuda:10.1-devel-ubuntu16.04`
in order to take advantages of tensorflow cuda capabilities when running the container with `singularity run --nv`  on cuda nodes 
as far I as know, `conda.miniconda` docker does not embeds cuda support 

here is the singularity recipe I wrote.

regards

NB tested with singularity 3.4.0 and singularity 3.5.3 
does not works with singularity-3.5.2 see #45 

[souporcell-2.0..recipe.recipe.txt](https://github.com/wheaton5/souporcell/files/4528838/souporcell-2.0.recipe.recipe.txt)

**Comment by wheaton5** [comment]
Note that souporcell 2.0 no longer uses tensorflow so if you are choosing things based on their tensorflow support don't bother. Other than that I'm not sure what our biggest glibc compatibility issues will be, maybe just htslib.

Other than that, do you see any downside to moving to this base?

**Comment by EricDeveaud** [comment]
every result of make in the container, ie htslib, santools, bcftools, bedtools. 
running cnetos6 gave me some headache with GLIBC_SYMBOL.... not found

freebayes is also subject to the problem. I have to build it inside the container in order to avoid problems.

I'm currently updating the singularity recipe to reflect that.

regrding the use of tensorflow, I did not dig really deeply in the code, I just got tensorflow insight from the check_module.py that contains an import for tensorflow
```
import tensorflow as tf
```
 
I will remove it from the recipe

Eric
